js/adapt-contrib-xapi.js

```diff
@@ -880,22 +910,38 @@ define([
               return nextType(new Error('\'xhr\' parameter is missing from callback'));
             }
 
-            if (xhr.status !== 200) {
+            if (xhr.status == 404) {
+              return nextType();
+            }
+
+            if (xhr.status != 200) {
               Adapt.log.warn('getState() failed for ' + activityId + ' (' + type + ')');
               return nextType(new Error('Invalid status code ' + xhr.status + ' returned from getState() call'));
             }
 
-            state[type] = JSON.parse(xhr.response);
+            var response;
+            var parseError;
+            try {
+              response = JSON.parse(xhr.response);
+            } catch (error) {
+              parseError = error;
+            }
+
+            if (parseError) {
+              return nextType(parseError);
+            }
+
+            if (_.isArray(response) && !_.isEmpty(response)) {
```

Comment:
I think forcing this to be an array is probably too strict.  It could be a JSON object, e.g. as a bookmark store.

Reply:
This was off the back of it expecting an array [here](https://github.com/adaptlearning/adapt-contrib-xapi/pull/24/files#diff-8055477379f7d61f34f1581108dfb75dR867) (I was seeing an error when I wasn't mocking it correctly when testing it locally)

I see what you mean though, course state is itself an object - I'll remove this